packages/dataprovider/src/buildQuery.ts

```diff
@@ -34,7 +36,21 @@ export const buildQueryFactory = (
       );
     }
 
-    const fragment = resourceView?.fragment ?? undefined;
+    let fragment: DocumentNode = undefined;
+    if (resourceView) {
+      // type union info is lost after compiling to JS
+      // however, we can check for existence of "one" or "many" fields in the fragment which would indicate DoubleFragment type
+      if ((resourceView.fragment as any).one) {
```

Comment:
🤔 I feel like I should be more careful with checking here. Typescript should warn the user that if they are specifying an object with `one` and `many`, they need to specify both with the correct type. But what about non-Typescript projects? There won't be any warnings that would stop the project from trying to be built and used, right?

Reply:
Can't check for `object` type, because `DocumentNode` is also an object, but checking for both properties should be fine? How about this?
- if both `one` and `many` are defined => it's correct and pick the correct fragment
- if none of them are defined => correct and it's a single fragment
- if only one of them is defined => wrong and throw an Error telling the user to either specify both or just use a single fragment?